needle-zendesk-connector/src/server/backend/connectors-backend.ts

```diff
@@ -0,0 +1,257 @@
+
+    descriptor.create.push({
+      id: articleFileId,
+      url: article.html_url,
+      type: "text/html",
+    });
+  }
+
+  await db.transaction(async (tx) => {
+    await tx.insert(zendeskConnectorsTable).values({
+      connectorId: connector.id,
+      subdomain: params.subdomain,
+      orgId: params.organizationId,
+      includeTickets: params.selectedTickets.length > 0,
+      includeArticles: params.selectedArticles.length > 0,
+    });
+
+    await handleDatabaseUpdates(
+      tx,
+      connector.id,
+      { create: params.selectedTickets, update: [], delete: [] },
+      { create: params.selectedArticles, update: [], delete: [] },
+      fileIdMap,
+    );
+  });
+
+  await publishConnectorRun(connector.id, descriptor);
+  return connector;
+}
+
+export async function runZendeskConnector(
+  { connectorId }: { connectorId: string },
+  session?: Session,
+) {
+  let accessToken = "";
+  if (session) {
+    const { credentials } = await getConnector(connectorId, session.id);
+    accessToken = credentials ?? "";
+  }
+
+  const connectorDetails = await getConnectorDetails(connectorId);
+  if (!connectorDetails) {
+    throw new Error(`No Zendesk connector found for ID: ${connectorId}`);
+  }
+
+  const currentFiles = await getCurrentFiles(connectorId);
+  const currentTickets = currentFiles.filter(
+    (f) => f.type === "ticket" || f.type === "comments",
+  ) as DbFile[];
+  const currentArticles = currentFiles.filter(
+    (f) => f.type === "article",
+  ) as DbFile[];
+
+  const zendeskService = createZendeskService(
+    accessToken,
+    connectorDetails.subdomain,
+  );
+  const { tickets: liveTickets, articles: liveArticles } =
+    await zendeskService.searchByOrganization(
+      connectorDetails.orgId.toString(),
+      {
+        fetchTickets: connectorDetails.includeTickets,
+        fetchArticles: connectorDetails.includeArticles,
+      },
+    );
+
+  const ticketsDiff = liveTickets
+    ? computeDiff(currentTickets, liveTickets.items)
+    : { create: [], update: [], delete: currentTickets };
+  const articlesDiff = liveArticles
+    ? computeDiff(currentArticles, liveArticles.items)
+    : { create: [], update: [], delete: currentArticles };
+
+  const descriptor: ConnectorRunDescriptor = {
+    create: [],
+    update: [],
+    delete: [],
+  };
+  const fileIdMap = new Map<string, string>();
+
+  // Process creates
+  for (const ticket of ticketsDiff.create) {
+    if ("subject" in ticket) {
+      const ticketFileId = createNeedleFileId();
+      const commentsFileId = createNeedleFileId();
+      fileIdMap.set(`ticket-${ticket.id}`, ticketFileId);
+      fileIdMap.set(`comments-${ticket.id}`, commentsFileId);
+
+      descriptor.create.push(
+        {
+          id: ticketFileId,
+          url: ticket.url,
+          type: "text/plain",
+        },
+        {
+          id: commentsFileId,
+          url: ticket.url.replace(".json", "/comments.json"),
```

Comment:
is this the correct url? 

In the API reference, it does not end in `.json`: https://developer.zendesk.com/api-reference/ticketing/tickets/ticket_comments/#list-comments

<img width="466" alt="Screenshot 2024-11-23 at 13 06 16" src="https://github.com/user-attachments/assets/45eb8c9d-0020-40a9-bf2d-6f55598c7fb5">

Reply:
oh i see now, got it 👍🏻 unusual that REST api endpoint ends with `.json` 😄 

<img width="738" alt="Screenshot 2024-11-25 at 18 07 04" src="https://github.com/user-attachments/assets/6f3cd034-3ec3-4482-926a-f9432fca6ed5">